src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -183,37 +183,19 @@
                 new AclKey( entitySetId ), Principals.getCurrentPrincipals(), READ_PERMISSION ) ) {
 
             Optional<Set<UUID>> entityKeyIds = ( selection == null ) ? Optional.empty() : selection.getEntityKeyIds();
-            Optional<Set<UUID>> propertyTypeIds = ( selection == null ) ? Optional.empty() : selection.getProperties();
-
-            final Set<UUID> allProperties = authzHelper.getAllPropertiesOnEntitySet( entitySetId );
-            final Set<UUID> selectedProperties = propertyTypeIds.orElse( allProperties );
-            checkState( allProperties.equals( selectedProperties ) || allProperties.containsAll( selectedProperties ),
-                    "Selected properties are not property types of entity set %s", entitySetId );
+            final var selectedProperties = getSelectedProperties( entitySetId, selection );
 
             final var entitySet = entitySetService.getEntitySet( entitySetId );
-            Set<UUID> normalEntitySetIds;
-            Map<UUID, Map<UUID, PropertyType>> authorizedPropertyTypesOfEntitySets;
-
-            checkState( entitySet != null, "Could not find entity set with id: " + entitySetId.toString() );
-
-            if ( entitySet.isLinking() ) {
-                normalEntitySetIds = Sets.newHashSet( entitySet.getLinkedEntitySets() );
-                checkState( !normalEntitySetIds.isEmpty(),
-                        "Linked entity sets are empty for linking entity set %s", entitySetId );
-
-                normalEntitySetIds.forEach( esId -> ensureReadAccess( new AclKey( esId ) ) );
-
-                authorizedPropertyTypesOfEntitySets = authzHelper
-                        .getAuthorizedPropertyTypesByNormalEntitySet( entitySet, selectedProperties, READ_PERMISSION );
-
-            } else {
-                normalEntitySetIds = Set.of( entitySetId );
-                authorizedPropertyTypesOfEntitySets = authzHelper
-                        .getAuthorizedPropertyTypes( normalEntitySetIds, selectedProperties, READ_PERMISSION );
-            }
+            checkState( entitySet != null, "Could not find entity set with id: %s", entitySetId );
```

Comment:
I think this check should happen before the usage of `getSelectedProperties`? Since the entitySetId is used within `getSelectedProperties`

Reply:
Yeah, that would make much more sense :) Thanks!